src/service.rs

```diff
@@ -612,23 +612,20 @@ impl CollectionManager {
     /// # Arguments:
     /// * `collection_type` - array of strings denoting the collection types
     /// * `options` - parameters to tune or optimize the fetch
-    pub fn list_multi<'a, I>(
+    pub fn list_multi<I>(
         &self,
         collection_types: I,
         options: Option<&FetchOptions>,
     ) -> Result<CollectionListResponse<Collection>>
     where
-        I: Iterator<Item = &'a str>,
+        I: IntoIterator,
```

Comment:
Correct me if I'm wrong, but this forces a clone at times, while the former is just a reference with a limited lifetime? Would this change be potentially worse?

Reply:
No, this change simply generalizes the `&str` item type to also allow passing `String` and such. The difference between this and the `ItemMetadata::set_*` functions is that we don't actually need a `String` in this function body, only a `&str` (to pass to `collection_type_to_uid`), so taking an argument that's "something like a `&str`" is fine.

Reply:
I'm talking about `IntoIterator` doesn't that take ownership?

Reply:
Every `Iterator<Item = I>` is also trivially `IntoIterator<Item = I>`, so for any type that was accepted previously, there is no change. This additionally allows directly passing collections without having to call `.into_iter()`/`.iter()` on them manually.

Reply:
Is it though? I though they are the same the other way around (so every IntoIterator is also Iterator). Anyhow, OK, not a biggie.

Reply:
No, see this impl: https://doc.rust-lang.org/core/iter/trait.IntoIterator.html#impl-IntoIterator-8

`Iterator` is the main trait behind rust iteration, `IntoIterator` is a trait that allows creating an `Iterator` from a collection (e.g. `Vec`), which is a potentially not-entirely-trivial operation. `IntoIterator` is also implemented for all `Iterator`s, but in that case, it's simply an identity function: https://doc.rust-lang.org/src/core/iter/traits/collect.rs.html#263-265

Reply:
Ah OK then.